src/CoWSwapEthFlow.sol

```diff
@@ -68,4 +77,47 @@ contract CoWSwapEthFlow is CoWSwapOnchainOrders, ICoWSwapEthFlow {
 
         orders[orderHash] = onchainData;
     }
+
+    /// @inheritdoc ICoWSwapEthFlow
+    function deleteOrder(EthFlowOrder.Data calldata order) external {
+        GPv2Order.Data memory cowSwapOrder = order.toCoWSwapOrder(
+            wrappedNativeToken
+        );
+        bytes32 orderHash = cowSwapOrder.hash(cowSwapDomainSeparator);
+
+        EthFlowOrder.OnchainData memory orderData = orders[orderHash];
+
+        if (
+            orderData.owner == EthFlowOrder.INVALIDATED_OWNER ||
+            orderData.owner == EthFlowOrder.NO_OWNER ||
+            (// solhint-disable-next-line not-rely-on-time
+            orderData.validTo >= block.timestamp &&
+                orderData.owner != msg.sender)
+        ) {
+            revert NotAllowedToDeleteOrder(orderHash);
+        }
+
+        orders[orderHash].owner = EthFlowOrder.INVALIDATED_OWNER;
+
+        bytes memory orderUid = new bytes(GPv2Order.UID_LENGTH);
+        orderUid.packOrderUidParams(
+            orderHash,
+            address(this),
+            cowSwapOrder.validTo
+        );
+        uint256 freedAmount = cowSwapOrder.sellAmount -
+            cowSwapSettlement.filledAmount(orderUid);
+
+        // Using low level calls to perform the transfer avoids setting arbitrary limits to the amount of gas used in a
+        // call. The drawback is allowing reentrancy from this point in the contract. Note that the current order was
+        // already invalidated and cannot be claimed again. TODO: ensure that this is robust under order updates when
+        // implemented.
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, ) = payable(orderData.owner).call{value: freedAmount}(
```

Comment:
This feels like a potential 🦶🔫 to watch out for.

Do you have any concerns of using `transfer` besides having to deal with access lists for smart contracts?

Reply:
It definitely is a foot gun. I liked the idea of supporting sending ETH to any smart contract, but the more I think about it the less I like this. I'm going to change this to use `transfer` as suggested, if we feel safe we can very easily remove it later. 

Since redeploying is cheap, we might want to play it safe at first and only add an arbitrary gas allowance only if we feel the need later. (Note: Alex noted how redeploying might be problematic for users who have long-standing ETH order, so redeploying has some cost.)

Reply:
The worst thing about this change is that it could lead to loss of user's funds.
If a user with a complicated contract creates an order then they won't be able to cancel the order anymore, which means that their ETH funds are lost. We can add heuristics in the backend to determine if a contract can receive ETH but it's complicated and not guaranteed to work for all cases.

For now I prefer to avoid reentrancy, but maybe before the audit we might want to reconsider this.

Reply:
>The worst thing about this change is that it could lead to loss of user's funds.
We can add heuristics in the backend to determine if a contract can receive ETH but it's complicated and not guaranteed to work for all cases.

I also see this as a risk. If my cool wallet contract has a receive function that is special and uses more than 2300 gas and I am using the ethflow contract, I might really shoot myself into the food.
I would stick with `call`.
If we wanna have the least 🦶🔫  possibilities, we might wanna go with `call` +  reentrcy guard.

Reply:
As this is the only thing blocking merge, I'll just merge as it is and implement a temporary reentrancy guard in another PR so we can continue the discussion there.